### PR TITLE
feat: Mongo image update and REST template

### PR DESCRIPTION
**Pull request by beatrizpvaz**
1. Update mongo image to work on Mac M chips
2. Rest template

### CODE REVIEW

File: package-lock.json
Comment (saulolso) [inline package-lock.json:1]: we don't need the package-lock.json, given we are using yarn. pls, remove this file.

File: src/posts/posts.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  findAll() {
+    return this.prisma.post.findMany({ where: { published: true } });
+  }
+
+  findOne(id: string) {
+    return this.prisma.post.findUnique({ where: { id } });
+  }
+
+  update(id: string, updatePostDto: UpdatePostDto) {
+    console.log('here,' + updatePostDto);
```
Comment (saulolso) [inline src/posts/posts.service.ts:23]: pls, remove console.log.

File: src/posts/posts.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      data: updatePostDto,
+    });
+  }
+  remove(id: string) {
+    return this.prisma.post.delete({ where: { id } });
+  }
+}
+
+// @Injectable()
+// export class ParseObjectIdPipe implements PipeTransform<any, ObjectId> {
```
Comment (saulolso) [inline src/posts/posts.service.ts:35]: leftover.

File: src/users/dto/create-user.dto.ts
Comment (saulolso) [inline src/users/dto/create-user.dto.ts:1]: let's keep naming consistent. atm, we have 3 different patterns: createPost.dto.ts, updatePostDto.dto.ts, and create-user.dto.ts.  Any of them is fine for me, but let's have just one naming pattern.

File: src/posts/posts.module.ts [hunk 3]
```diff
@@ -1,7 +1,11 @@
 import { Module } from '@nestjs/common';
+import { PostsController } from './posts.controller';
+import { PostService } from './posts.service';
+import { PrismaModule } from 'nestjs-prisma';
 
 @Module({
-  imports: [],
-  providers: [],
+  imports: [PrismaModule],
```
Comment (saulolso) [inline src/posts/posts.module.ts:7]: We are declaring PrismaModule as a global module on the `AppModule.ts`. So I think it's OK to remove this import.

File: src/common/validators/validators.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { BadRequestException, Injectable, PipeTransform } from '@nestjs/common';
+import { ObjectId } from 'mongodb';
```
[the remark this answers is not in the document]
Comment (beatrizpvaz) [reply src/common/validators/validators.ts]: Totally agree! I actually used something like this in my Python projects

**Review by saulolso — APPROVED** [review]

**Pull request merged**